draft-nygren-dnsop-svcb-httpssvc.md

```diff
@@ -494,212 +489,191 @@ When SvcRecordType is the ServiceForm, the combination of
+this value.)
 
 ### SvcFieldPriority  {#svcfieldpri}
 
 As RRs within an RRSET are explicitly unordered collections, the
-SvcFieldPriority value is introduced to indicate priority.
+SvcFieldPriority value serves to indicate priority.
 SVCB RRs with a smaller SvcFieldPriority value SHOULD be given
 preference over RRs with a larger SvcFieldPriority value.
 
 When receiving an RRSET containing multiple SVCB records with the
 same SvcFieldPriority value, clients SHOULD apply a random shuffle within a
-priority level to the records before using them, to ensure randomized
+priority level to the records before using them, to ensure uniform
 load-balancing.
 
 
-## General SvcParamKeys for use in SvcFieldValue
-
-A few SvcParamKeys are defined here for general use.
+# Client behavior {#client-behavior}
 
-### SvcParamKey: alpn
+An SVCB-aware client resolves an origin by attempting to determine
+the preferred SvcFieldValue and IP addresses and SvcFieldValue for its service, using the
+following procedure:
 
-TODO: should we call this "alpn" or "proto"?  Alt-Svc calls it
-protocol-id so perhaps "proto" is better and easier for most
-people to understand?
+1. Issue parallel AAAA/A and SVCB queries for the name HOST.
+   The answers for these may or may not include CNAME pointers
+   before reaching one or more of these records.
 
-This the "alpn" SvcParamKey defines the Application Layer Protocol
-(ALPN, as defined in {{!RFC7301}) supported by this alternative service.
+2. If an SVCB record of AliasForm SvcRecordType is returned for HOST,
+   clients MUST loop back to step 1 replacing HOST with SvcDomainName,
+   subject to loop detection heuristics.
 
-The presentation format of the SvcParamValue is a UTF-8 string
-identifying the protocol.  The wire format of SvcParamValue
-is the corresponding set of octet values identifying the protocol.
+3. If one or more SVCB records of ServiceForm SvcRecordType are returned
+   for HOST, clients should select the highest-priority option with
+   acceptable parameters, and resolve AAAA and/or A records for its
+   SvcDomainName if they are not already available.  These are the
+   preferred SvcFieldValue and IP addresses.  If connection fails, the
+   client MAY try to connect using values from a lower-priority record.
 
-Clients MUST ignore SVCB RRs where the "alpn" SvcParamValue
-is unknown or unsupported.
+4. If an SVCB record for HOST does not exist, the received AAAA and/or A
+   records are the preferred IP addresses, and there is no preferred
+   SvcFieldValue.
 
-### SvcParamKey: port
+This procedure does not rely on any recursive or authoritative server to
+comply with this specification or have any awareness of SVCB.
 
-This the "port" SvcParamKey defines the TCP or UDP port
-that should be used to contact this alternative service.
+When selecting between AAAA and A records to use, clients may
+use an approach such as {{!HappyEyeballsV2=RFC8305}}
```

Comment:
Thanks for the reference!

As a nit, missing a period at the end of the sentence:
```suggestion
use an approach such as {{!HappyEyeballsV2=RFC8305}}.
```